src/GraphQL/Types/Composite/AutoRegisteringHelper.cs

```diff
@@ -224,5 +224,143 @@ internal static LambdaExpression GetParameterExpression(Type parameterType, Quer
+                        parameterInfo.ParameterType,
+                        queryArgument ?? throw new InvalidOperationException("Invalid response from ConstructQueryArgument: queryArgument and expression cannot both be null"));
+                    expressions.Add(expression);
+                }
+                var memberInstanceExpression = BuildMemberInstanceExpression(methodInfo);
+                if (IsObservable(methodInfo.ReturnType))
+                {
+                    var resolver = new SourceStreamMethodResolver(methodInfo, memberInstanceExpression, expressions);
+                    fieldType.Resolver = resolver;
+                    fieldType.StreamResolver = resolver;
+                }
+                else
+                {
+                    var resolver = new MemberResolver(methodInfo, memberInstanceExpression, expressions);
+                    fieldType.Resolver = resolver;
+                    fieldType.StreamResolver = null;
+                }
+                fieldType.Arguments = queryArguments;
+            }
+            else if (memberInfo is FieldInfo fieldInfo)
+            {
+                var resolver = new MemberResolver(fieldInfo, BuildMemberInstanceExpression(memberInfo));
+                fieldType.Arguments = null;
+                fieldType.Resolver = resolver;
+                fieldType.StreamResolver = null;
+            }
+            else if (memberInfo == null)
+            {
+                throw new ArgumentNullException(nameof(memberInfo));
+            }
+            else
+            {
+                throw new ArgumentOutOfRangeException(nameof(memberInfo), "Member must be a field, property or method.");
+            }
+        }
+
+        /// <summary>
+        /// Determines if the type is an <see cref="IObservable{T}"/> or task that returns an <see cref="IObservable{T}"/>.
+        /// </summary>
+        private static bool IsObservable(Type type)
+        {
+            if (!type.IsGenericType)
+                return false;
+
+            var g = type.GetGenericTypeDefinition();
+            if (g == typeof(IObservable<>))
+                return true;
+            if (g == typeof(Task<>) || g == typeof(ValueTask<>))
+                return IsObservable(type.GetGenericArguments()[0]);
+            return false;
+        }
+
+        /// <summary>
+        /// Returns a list of properties, methods or fields that should have fields created for them.
+        /// <br/><br/>
+        /// Unless overridden, returns a list of public instance readable properties and public instance methods
+        /// that do not return <see langword="void"/> or <see cref="Task"/>
+        /// including properties and methods declared on inherited classes.
+        /// </summary>
+        internal static IEnumerable<MemberInfo> GetRegisteredMembersForOutput<TSourceType>(Expression<Func<TSourceType, object?>>[]? excludedProperties)
```

Comment:
This method consolidates code between `AutoRegisteringInterfaceGraphType` and `AutoRegisteringObjectGraphType`.  As can be seen here, typically there is no shared executable code between the two implementations, this consolidation does allow interfaces to be constructed with `AutoRegisteringObjectGraphType` and classes with `AutoRegisteringInterfaceGraphType`.  There seems to be no reason not to design it this way, and it does allow for interfaces to be used to separate the graph attributes from GraphQL object graph types.